lib/features/manage_account/data/datasource_impl/notification_datasource_impl.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:app_settings/app_settings.dart';
+import 'package:jmap_dart_client/jmap/core/user_name.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:tmail_ui_user/features/manage_account/data/datasource/notification_datasource.dart';
+import 'package:tmail_ui_user/main/exceptions/exception_thrower.dart';
+import 'package:tmail_ui_user/main/permissions/notification_permission_service.dart';
+
+class NotificationDataSourceImpl implements NotificationDataSource {
+  final NotificationPermissionService _permissionService;
+  final ExceptionThrower _exceptionThrower;
+
+  NotificationDataSourceImpl(
+    this._permissionService,
+    this._exceptionThrower);
+
+  @override
+  Future<bool> getNotificationSetting(UserName userName) {
+    return Future.sync(() async {
+      _permissionService.setNotificationGroupId(userName.value);
```

Comment:
why inside a get function we have a set function?

Reply:
we should find better way to check notification group enable. Why not a dedicated function to notification settings?

Reply:
Updated